src/component.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable, IOnChangesObject } from 'angular';
+
+/**
+ * Valid properties to make an angularJS component. The `controller` and `controllerAs` properties
+ * have been removed since we are trying to wrap a component to make it look more like an Angular
+ * component.
+ */
+interface INgComponentOptions {
+  selector: string;
+  template?: string | Injectable<(...args: any[]) => string>;
+  templateUrl?: string | Injectable<(...args: any[]) => string>;
+  bindings?: {[boundProperty: string]: string};
+  transclude?: boolean | {[slot: string]: string};
+  require?: {[controller: string]: string};
+}
+
+/**
+ * Attaches `__component` to the clazz object. If it already has one, then it overrides the
+ * properties it needs to.
+ *
```

Comment:
I'll fix this comment. The two underscores? Well, I wanted something that angular won't use, they already use `$inject`, I thought maybe using `$component` but that's kinda reserved for them so I just went for the `__`. I'm open to other prefix. Remember that this property is attached to the class constructor, i.e. is a static property.